pkg/agent/multicast/mcast_discovery.go

```diff
@@ -120,16 +123,17 @@ func (s *IGMPSnooper) validate(event *mcastGroupEvent) (bool, error) {
 		return true, nil
 	}
 	if event.iface.Type == interfacestore.ContainerInterface {
-		item, err := s.Validator.Validate(event.iface.PodName, event.iface.PodNamespace, event.group)
+		item, err := s.validator.Validate(event.iface.PodName, event.iface.PodNamespace, event.group)
```

Comment:
Not your code, but hard to associate the name Validate() with multicast in its implementor of NetworkPolicy Controller.

Reply:
Maybe GetIGMPReportRuleAction()? @liu4480